### PR TITLE
Authentication module duplicated code refactor

### PR DESCRIPTION
**Pull request by Kami**

### CODE REVIEW

File: st2common/st2common/hooks.py [hunk 1]
```diff
@@ -106,24 +104,5 @@ def _validate_token(request):
 
         token_in_headers = headers.get('X_Auth_Token', None)
```
Comment (manasdk) [inline st2common/st2common/hooks.py]: can they be the same strings? are there some weird restrictions that would prevent doing so.

Ideally, X-Auth-Token

File: st2common/st2common/middleware/auth.py [hunk 2]
```diff
@@ -77,20 +76,8 @@ def _validate_token(self, env):
         token_in_headers = env.get('HTTP_X_AUTH_TOKEN', None)
```
Comment (manasdk) [inline st2common/st2common/middleware/auth.py]: huh?

File: st2common/st2common/util/auth.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+    :type token_in_headers: ``str``
+
+    :param token_in_query_params: Authentication token provided via query params.
+    :type token_in_query_params: ``str``
+
+    :return: TokenDB object on success.
+    :rtype: :class:`.TokenDB`
+    """
+    if not token_in_headers and not token_in_query_params:
+        LOG.audit('Token is not found in header or query parameyers.')
```
Comment (manasdk) [inline st2common/st2common/util/auth.py]: should be an exception. Does putting in audit provide much value?

Comment (manasdk) [inline st2common/st2common/util/auth.py]: s/parameyers/parameters

File: st2common/st2common/util/auth.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+        raise exceptions.TokenNotProvidedError('Token is not provided.')
+
+    if token_in_headers:
+        LOG.audit('Token provided in headers')
+
+    if token_in_query_params:
+        LOG.audit('Token provided in query parameters')
```
Comment (manasdk) [inline st2common/st2common/util/auth.py]: all these messages at audit would only litter the audit log. Not sure if they add much value.

File: st2common/st2common/hooks.py (hunk 1 above)
Comment (Kami) [inline st2common/st2common/hooks.py]: I didn't touch or change the key names - wsgi handling code in webob just does some weird key renaming during look-ups inside __getattr__.

I can test if X-Auth-Token also works and if does, use that to make it more clear and obvious.

File: st2common/st2common/util/auth.py [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+
+    token_string = token_in_headers or token_in_query_params
+    token = Token.get(token_string)
+
+    if token.expiry <= isotime.add_utc_tz(datetime.datetime.utcnow()):
+        # TODO: purge expired tokens
+        LOG.audit('Token "%s" has expired.' % (token_string))
+        raise exceptions.TokenExpiredError('Token has expired.')
+
+    LOG.audit('Token "%s" is validated.' % (token_string))
```
Comment (manasdk) [inline st2common/st2common/util/auth.py]: Is it ok to log this? t is as good as a password.

File: st2common/st2common/middleware/auth.py (hunk 2 above)
Comment (Kami) [inline st2common/st2common/middleware/auth.py]: This is an environment variable name set by wsgi - as above, I didn't touch or change key name.

File: st2common/st2common/util/auth.py (hunk 4 above)
Comment (Kami) [inline st2common/st2common/util/auth.py]: Same here - this is an existing code, I didn't touch it, but I can remove it.

**Comment by manasdk** [comment]
+1

Comments mostly cosmetic. Feel free to pick and choose what you would like to address.

File: st2common/st2common/util/auth.py (hunk 5 above)
Comment (Kami) [inline st2common/st2common/util/auth.py]: Same here, this is an existing code.

I do agree with you and I can change it so we log the Token object id instead.

**Comment by Kami** [comment]
Changed the code to log id instead of value - 30d1605e3e06f49036d169f40967db437f8c1ae6

**Pull request merged**